src/context/WalletProvider/MetaMask/components/Connect.tsx

```diff
@@ -91,13 +90,20 @@ export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
     setLoading(false)
   }
 
+  // This constructs the MetaMask deep-linking target from the currently-loaded
+  // window.location. The port will be blank if not specified, in which case it
+  // should be omitted.
+  const mmDeeplinkTarget = [window.location.hostname, window.location.port]
+    .filter(x => !!x)
+    .join(':')
+
```

Comment:
Maybe we ought to do some regexp validation on the target URL here so that we can catch errors and fail appropriately. Otherwise, LGTM.

Reply:
IMO if those are wrong it's the browser's fault. The spec is defined in terms of strings and concatenation. Only edge case I can think of might be IPV6 addresses and needing `[addr]:port` syntax? I bet MM would choke on those though.